src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,28 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> {
```

Comment:
Before processing a request, check if the userId is already in the cache. If yes, return the cached result immediately.
After processing a request, store the result in the cache.
I would better move this logic to a separate `CompletableFuture`.